spec/dds_md5_subscriber_spec.rb

```diff
@@ -0,0 +1,54 @@
+require_relative '../dds_md5_subscriber'
+require 'bunny-mock'
+
+describe DdsMd5Subscriber do
+  let(:file_version_id) { 'FILE_VERSION_ID' }
+
+  it { expect(described_class).to include(Sneakers::Worker) }
+  it { expect(ENV['TASK_QUEUE_NAME']).not_to be_nil }
+  it { expect(subject.queue.name).to eq(ENV['TASK_QUEUE_NAME']) }
+
+  describe '#work' do
+    let(:mock_reporter) { instance_double('DdsMd5Reporter') }
+    it { is_expected.to respond_to(:work) }
+    let(:method) { subject.work(file_version_id) }
+    let(:ack) { subject.ack! }
+    let(:reject) { subject.reject! }
+
+    context 'successful report' do
+      it {
+        Sneakers.configure(connection: BunnyMock.new)
+        expect(DdsMd5Reporter).to receive(:new)
+          .with(
+            file_version_id: file_version_id,
+            user_key: ENV['USER_KEY'],
+            agent_key: ENV['AGENT_KEY'],
+            dds_api_url: ENV['DDS_API_URL']
+          ).and_return(mock_reporter)
+        expect(mock_reporter).to receive(:report_md5)
```

Comment:
To dry out the tests a bit, the 8 lines above could go in a `before(:example)` block and reused for both success and error contexts. To generalize the `:report_md5` mock, use a return block:
```
expect(mock_reporter).to receive(:report_md5) { report_md5_response }
```
And just set the `report_md5_response` to `nil` or `raise expected_error` in each context.

Reply:
I will try this again, but for some reason I tried running that in a before block at some point and the test failed. When I moved it into the it block, it passed. Not sure why.

Reply:
done